metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py

```diff
@@ -541,6 +565,15 @@ def _schema_aware_fuzzy_column_resolve(
         ) from e
     logger.debug("Qualified sql %s", statement.sql(pretty=True, dialect=dialect))
 
+    # Try to figure out the types of the output columns.
+    try:
+        statement = sqlglot.optimizer.annotate_types.annotate_types(
+            statement, schema=sqlglot_db_schema
+        )
+    except sqlglot.errors.OptimizeError as e:
+        # This is not a fatal error, so we can continue.
+        logger.debug("sqlglot failed to annotate types: %s", e)
```

Comment:
Could this be slow? I think it'd be nice to only do this if a config option is specified

Reply:
this step should be pretty fast